homeassistant/components/cisco_ios/device_tracker.py

```diff
@@ -115,26 +114,23 @@ def _get_arp_data(self):
             )
 
             # Find the hostname
-            initial_line = cisco_ssh.before.decode("utf-8").splitlines()
+            initial_line = (cisco_ssh.before or "").splitlines()
             router_hostname = initial_line[len(initial_line) - 1]
             router_hostname += "#"
             # Set the discovered hostname as prompt
-            regex_expression = f"(?i)^{router_hostname}".encode()
-            cisco_ssh.PROMPT = re.compile(regex_expression, re.MULTILINE)
+            cisco_ssh.PROMPT = f"(?i)^{router_hostname}"
             # Allow full arp table to print at once
```

Comment:
Both options (`str` and the compiled regex) will technically work. It's typed as just a string in the types package so I used that. Might be a slight inaccuracy in the types.